docs/spec/directives.rst

```diff
@@ -0,0 +1,264 @@
+
+``cast()``
+----------
+
+Occasionally the type checker may need a different kind of hint: the
+programmer may know that an expression is of a more constrained type
+than a type checker may be able to infer.  For example::
+
+  from typing import cast
+
+  def find_first_str(a: list[object]) -> str:
+      index = next(i for i, x in enumerate(a) if isinstance(x, str))
+      # We only get here if there's at least one string in a
+      return cast(str, a[index])
+
+Some type checkers may not be able to infer that the type of
+``a[index]`` is ``str`` and only infer ``object`` or ``Any``, but we
+know that (if the code gets to that point) it must be a string.  The
+``cast(t, x)`` call tells the type checker that we are confident that
+the type of ``x`` is ``t``.  At runtime a cast always returns the
+expression unchanged -- it does not check the type, and it does not
+convert or coerce the value.
+
+Casts differ from type comments (see the previous section).  When using
+a type comment, the type checker should still verify that the inferred
+type is consistent with the stated type.  When using a cast, the type
+checker should blindly believe the programmer.  Also, casts can be used
+in expressions, while type comments only apply to assignments.
+
+``TYPE_CHECKING``
+-----------------
+
+Sometimes there's code that must be seen by a type checker (or other
+static analysis tools) but should not be executed.  For such
+situations the ``typing`` module defines a constant,
+``TYPE_CHECKING``, that is considered ``True`` during type checking
+(or other static analysis) but ``False`` at runtime.  Example::
+
+  import typing
+
+  if typing.TYPE_CHECKING:
+      import expensive_mod
+
+  def a_func(arg: 'expensive_mod.SomeClass') -> None:
+      a_var: expensive_mod.SomeClass = arg
+      ...
+
+(Note that the type annotation must be enclosed in quotes, making it a
+"forward reference", to hide the ``expensive_mod`` reference from the
+interpreter runtime.  In the variable annotation no quotes are needed.)
+
+This approach may also be useful to handle import cycles.
+
+``@no_type_check``
+------------------
+
+To mark portions of the program that should not be covered by type
+hinting, you can use the ``@typing.no_type_check`` decorator on a class or function.
+Functions with this decorator should be treated as having
+no annotations.
+
+Version and platform checking
+-----------------------------
+
+Type checkers are expected to understand simple version and platform
+checks, e.g.::
+
+  import sys
+
+  if sys.version_info >= (3, 12):
+      # Python 3.12+
+  else:
+      # Python 3.11 and lower
+
+  if sys.platform == 'win32':
+      # Windows specific definitions
+  else:
+      # Posix specific definitions
+
+Don't expect a checker to understand obfuscations like
+``"".join(reversed(sys.platform)) == "xunil"``.
+
+``@deprecated``
+---------------
+
+(Originally specified in :pep:`702`.)
+
+The :py:func:`warnings.deprecated`
```

Comment:
```suggestion
The :class:`warnings.deprecated`
```
As this is now implemented as a class?

Reply:
It's not documented as a class: https://docs.python.org/3.13/library/warnings.html#warnings.deprecated

Reply:
Ah yes makes sense, it isn't auto-generated docs